protocols/dcutr/examples/client.rs

```diff
@@ -0,0 +1,275 @@
+        identify: Identify::new(IdentifyConfig::new(
+            "/TODO/0.0.1".to_string(),
+            local_key.public(),
+        )),
+        dcutr: dcutr::behaviour::Behaviour::new(),
+    };
+
+    let mut swarm = SwarmBuilder::new(transport, behaviour, local_peer_id)
+        .dial_concurrency_factor(10_u8.try_into().unwrap())
+        .build();
+
+    swarm
+        .listen_on(
+            Multiaddr::empty()
+                .with("0.0.0.0".parse::<Ipv4Addr>().unwrap().into())
+                .with(Protocol::Tcp(0)),
+        )
+        .unwrap();
+
+    // Wait to listen on localhost.
+    block_on(async {
+        let mut delay = futures_timer::Delay::new(std::time::Duration::from_secs(1)).fuse();
+        loop {
+            futures::select! {
+                            event = swarm.next() => {
+                                match event.unwrap() {
+                                    SwarmEvent::NewListenAddr { address, .. } => {
+            info!("Listening on {:?}", address);
+                                    }
+                                    event => panic!("{:?}", event),
+                                }
+                            }
+                            _ = delay => {
+                                break;
+                            }
+                        }
+        }
+    });
+
+    match opts.mode {
+        Mode::Dial => {
+            swarm.dial(opts.relay_address.clone()).unwrap();
+        }
+        Mode::Listen => {
+            swarm
+                .listen_on(opts.relay_address.clone().with(Protocol::P2pCircuit))
+                .unwrap();
+        }
+    }
+
+    // Wait till connected to relay to learn external address.
+    block_on(async {
+        loop {
+            match swarm.next().await.unwrap() {
+                SwarmEvent::NewListenAddr { .. } => {}
+                SwarmEvent::Dialing { .. } => {}
+                SwarmEvent::ConnectionEstablished { .. } => {}
+                SwarmEvent::Behaviour(Event::Ping(_)) => {}
+                SwarmEvent::Behaviour(Event::Relay(_)) => {}
+                SwarmEvent::Behaviour(Event::Identify(IdentifyEvent::Sent { .. })) => {}
+                SwarmEvent::Behaviour(Event::Identify(IdentifyEvent::Received {
+                    info: IdentifyInfo { observed_addr, .. },
+                    ..
+                })) => {
+                    info!("Observed address: {:?}", observed_addr);
+                    break;
+                }
+                event => panic!("{:?}", event),
+            }
+        }
+    });
+
+    if matches!(opts.mode, Mode::Dial) {
+        swarm
+            .dial(
+                opts.relay_address
+                    .clone()
+                    .with(Protocol::P2pCircuit)
+                    .with(Protocol::P2p(opts.remote_peer_id.unwrap().into())),
+            )
+            .unwrap();
+    }
+
+    block_on(futures::future::poll_fn(move |cx: &mut Context<'_>| {
```

Comment:
Why `poll_fn` here and and `async` loop above?

Reply:
Good point. Replaced with `async` in 128d8015d6e6dc68c824292d661f1ee62029fb49.